clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -330,27 +332,31 @@ public int hashCode() {
         private int messagesRetained = 0;
         private int bytesRetained = 0;
         private long maxOffset = -1L;
+        // Retains invariant `minOffset` = the baseOffset of the first batch of retained records
+        // or -1 if no batches are retained
+        private long minOffset = -1L;
```

Comment:
It's more verbose, but perhaps we should choose a more explicit name like `baseOffsetOfFirstBatch`. Then we could either get rid of the comment or move to the getter for this field (as a javadoc).